tests/hooks/after_prepare.js

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env node
+
+/*
+*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*/
+
+var path = require('path');
+var fs = require('fs');
+
+module.exports = function(context) {
+    function main() {
+        // get the file transfer server address from the specified variables
+        var defaultFileTransferServerAddress = getDefaultFileTransferServerAddress(context);
```

Comment:
I think you can optimize this a bit by calling `getDefaultFileTransferServerAddress()` only when server address is not set explicitly:

```js
var fileTransferServerAddress = getFileTransferServerAddress(context) ||
                                getDefaultFileTransferServerAddress(context);
```